### PR TITLE
Avoid NullPointerException on showing toast

### PR DESCRIPTION
**Pull request by KeithYokoma**
- Avoid calling getContext() after fragment detached

### CODE REVIEW

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/SessionFeedbackFragment.java [hunk 1]
```diff
@@ -100,11 +100,15 @@ public void onClickSubmitFeedback() {
 
     public void onSubmitSuccess() {
         // TODO: show success action
+        if (isDetached())
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/SessionFeedbackFragment.java]: Let me ask you one thing.
Can we clear `Disposable` when Fragment is detached?
I mean like this.

```java
     @Override
     public void onClickSubmitFeedback() {
         SessionFeedback sessionFeedback = new SessionFeedback(sessionId);
         disposable = viewModel.submitSessionFeedback(sessionFeedback)
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(success -> onSubmitSuccess(), failure -> onSubmitFailure());

        compositeDisposable.add(disposable);
     }

    @Override
    public void onDetach() {
        super.onDetach();
        compositeDisposable.clear();
    }
```

**Comment by konifar** [comment]
@KeithYokoma ping just in case 🙇

**Comment by KeithYokoma** [comment]
OK I will do it!

**Comment by konifar** [comment]
Thanks!!

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/SessionFeedbackFragment.java [hunk 2]
```diff
@@ -90,6 +90,12 @@ public void onDestroy() {
...
     @Override
+    public void onDetach() {
+        super.onDetach();
+        compositeDisposable.clear();
+    }
+
+    @Override
     public void onClickSubmitFeedback() {
         SessionFeedback sessionFeedback = new SessionFeedback(sessionId);
         viewModel.submitSessionFeedback(sessionFeedback)
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/SessionFeedbackFragment.java]: Can you call `compositeDisposable.add(disposable)`?

```java
         disposable = viewModel.submitSessionFeedback(sessionFeedback)
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(success -> onSubmitSuccess(), failure -> onSubmitFailure());

        compositeDisposable.add(disposable);
```

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/SessionFeedbackFragment.java [hunk 3]
```diff
@@ -100,11 +106,15 @@ public void onClickSubmitFeedback() {
...
     public void onSubmitSuccess() {
         // TODO: show success action
+        if (isDetached())
+            return;
         Toast.makeText(getContext(), "submit success", Toast.LENGTH_SHORT).show();
     }
 
     public void onSubmitFailure() {
         // TODO: show failure action
+        if (isDetached())
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/SessionFeedbackFragment.java]: You called `compositeDisposable.clear()` in `onDeach()`.
Is this `isDetached()` necessary here?

**Comment by KeithYokoma** [comment]
Sorry for late but I've applied feedback!

**Comment by konifar** [comment]
LGTM! 💯 Thanks!

**Pull request merged**